### ISSUE TITLE
Drop tslint in favor of eslint.

### ISSUE THREAD

**Issue by mshima** [jhipster/generator-jhipster#12764]
##### **Overview of the feature request**

Migrate to eslint.

##### **Motivation for or Use Case**

```
npm WARN deprecated tslint@6.1.2: TSLint has been deprecated in favor of ESLint. Please see https://github.com/palantir/tslint/issues/4534 for more information.
```

##### **Related issues or PR**

-   [ ] Checking this box is mandatory (this is just to show you read everything)

**Comment by pascalgrimaud** [comment]
The work has started for Vue.js by @atomfrede here: https://github.com/jhipster/jhipster-vuejs/pull/601 but was not finished.

About the other front, I think it's already done but I'm not sure

**Comment by atomfrede** [comment]
I try to integrate the changes in the main generator this weekend. I am pretty sure for angular it is done already not sure about react though.

**Comment by mshima** [comment]
@pascalgrimaud @atomfrede every framework still depends on tslint.
I plan to migrate angular soon.

**Comment by atomfrede** [comment]
Good to know. So we need someone to do the react part 😁

**Comment by vishal423** [comment]
If I remember correctly, last time the migration was done for `angular` and `react`. Though, `angular` code still has dependencies on `tslint` as the angular specific rules were not migrated to eslint by that time. If that transition path is available on angular, then, we can go ahead and migrate those rules to eslint and can remove the `tslint` dependencies

**Comment by mshima** [comment]
I was wrong react doesn't depends on tslint.
Vue and angular are missing.
- angular: https://github.com/angular-eslint/angular-eslint#migrating-from-codelyzer-and-tslint

**Comment by kaidohallik** [comment]
@mshima I would like to handle Angular part if this ok for you.

https://github.com/angular-eslint/angular-eslint is currently in beta but they want to release 1.0.0 this year. As `tslint` is deprecated then I think it's ok to migrate to `angular-eslint` already now. I have used this beta version of `angular-eslint` and for me this is working fine.

I would like to upgrade `@typescript-eslint` from version 2 to version 4. Our current Angular code doesn't meet to recommended rules of version 4, some rule violations are not easy to fix and those rules I turn off in config file.

I also would like to apply some additional rules in addition to the recommended ones.

**Comment by mshima** [comment]
@kaidohallik sure. Just care to not conflict with angular cli PR too much. Or use that branch already.
We should merge it ASAP.

**Comment by kaidohallik** [comment]
#12763 dropped `eslint-config-jhipster` dependency but added common ESLint config `.eslint.common.js` for Angular and React. I would like to remove this common config also and handle ESLint in each framework separately.

Currently I'm trying to upgrade ESLint dependencies and this means this common file needs changes. It means I'm basically forced to do this upgrade in parallel for Angular and React which makes this task very time consuming.

**Comment by mshima** [comment]
@kaidohallik have you tried https://github.com/typescript-eslint/tslint-to-eslint-config?
I used it working once. It creates a big rule set but I think it’s easier.

About the shared config, I just copied from the other project. It would be great to have shared rules, but not sure it’s a good idea since every client framework is different.

**Comment by kaidohallik** [comment]
Before migrating rules I want to update ESLint dependencies to benefit from latest rules.

I made #12825 to drop shared config because it's easier to maintain for us and generated code is simpler and easier to understand.

**Comment by DanielFran** [comment]
@atomfrede @pascalgrimaud It seems that now this is only for Vue?
@atomfrede Any progress on this?

**Comment by atomfrede** [comment]
Hope to find some time tomorrow to revisit the current PR in the blueprint and migrate a first version to the main generator.
